substrate/frame/balances/src/lib.rs

```diff
@@ -256,7 +258,11 @@ pub mod pallet {
 
 		/// The overarching hold reason.
 		#[pallet::no_default_bounds]
-		type RuntimeHoldReason: Parameter + Member + MaxEncodedLen + Copy;
+		type RuntimeHoldReason: Parameter + Member + MaxEncodedLen + Copy + VariantCount;
+
+		/// The overarching freeze reason.
+		#[pallet::no_default_bounds]
+		type RuntimeFreezeReason: VariantCount;
```

Comment:
Shouldn't we add `Parameter + Member + MaxEncodedLen + Copy +` for `RuntimeFreezeReason` as well?

Reply:
There's nothing in this pallet's use of `RuntimeFreezeReason` that would mandate it to be, why should we add them?